extensions/amp-lightbox/0.1/amp-lightbox.js

```diff
@@ -189,36 +190,54 @@ class AmpLightbox extends AMP.BaseElement {
         .then(() => this.finalizeOpen_());
   }
 
+  /**
+   * Any child of the lightbox with the autofocus attribute should be focused
+   * after the lightbox opens.
+   * @private
+   */
+  handleAutofocus_() {
+    const autofocusElement = this.container_.querySelector('[autofocus]');
+    if (autofocusElement) {
+      tryFocus(autofocusElement);
+    }
+  }
+
+  /**
+   * @private
+   */
   finalizeOpen_() {
     if (this.isScrollable_) {
       st.setStyle(this.element, 'webkitOverflowScrolling', 'touch');
     }
-    this.mutateElement(() => {
-      st.setStyles(this.element, {
-        display: '',
-        opacity: 0,
-        // TODO(dvoytenko): use new animations support instead.
-        transition: 'opacity 0.1s ease-in',
-      });
-      Services.vsyncFor(this.win).mutate(() => {
-        st.setStyle(this.element, 'opacity', '');
-      });
-    }).then(() => {
-      const container = dev().assertElement(this.container_);
-      if (!this.isScrollable_) {
-        this.updateInViewport(container, true);
-      } else {
-        this.scrollHandler_();
-        this.updateChildrenInViewport_(this.pos_, this.pos_);
-      }
-      // TODO: instead of laying out children all at once, layout children based
-      // on visibility.
-      this.element.addEventListener('transitionend', this.boundReschedule_);
-      this.element.addEventListener('animationend', this.boundReschedule_);
-      this.scheduleLayout(container);
-      this.scheduleResume(container);
-      this.triggerEvent_(LightboxEvents.OPEN);
+
+    // This should be in a mutateElement block, but iOS focus doesn't work
+    // if triggered asynchronously.
+    st.setStyles(this.element, {
+      display: '',
+      opacity: 0,
+      // TODO(dvoytenko): use new animations support instead.
+      transition: 'opacity 0.1s ease-in',
     });
+    Services.vsyncFor(this.win).mutate(() => {
```

Comment:
@jridgewell can we put this in a `mutateElement` block? I don't actually know why this still uses a vsync. I'm assuming that the original intention here was to queue this into the next rAF mutation. 

So instead of calling `mutateElement` with an empty block, could we do this instead of the current vsync? 
```
this.mutateElement(() => {
    st.setStyle(this.element, 'opacity', '');
});